src/main/groovy/com/readytalk/gradle/plugins/ci/CiPublishingPlugin.groovy

```diff
@@ -43,47 +45,43 @@ class CiPublishingPlugin implements Plugin<Project>, PluginUtils {
     //TODO: Optional default publications for java/web components
   }
 
-  void configurePublishing() {
+  private void configurePublishing() {
+    configureBintrayPublishing()
+    configureArtifactoryPublishing()
+    configurePluginPublishing()
+    configureMavenPublishing()
+    configureIvyPublishing()
+    //TODO: Snapshot verification
+  }
+
+  private void configurePluginPublishing() {
     project.with {
-      withPluginId('com.jfrog.bintray') {
-        //TODO: Expand bintray support
-        logger.info('Bintray support limited to publish wiring only')
-        tasks.getByName(CiLifecyclePlugin.PUBLISH_TASK).dependsOn tasks.getByName('bintrayUpload')
-        tasks['bintrayUpload'].mustRunAfter 'build'
+      plugins.withId('com.gradle.plugin-publish') {
+        tasks.getByName(PublishingPlugin.PUBLISH_LIFECYCLE_TASK_NAME).dependsOn tasks.getByName('publishPlugins')
+        tasks['publishPlugins'].mustRunAfter 'build'
       }
-
-      withPluginId('com.jfrog.artifactory') {
-        //Ensure publishing goes through artifactoryPublish when using artifactory plugin
-        tasks[CiLifecyclePlugin.PUBLISH_TASK].dependsOn tasks['artifactoryPublish']
-        tasks['artifactoryPublish'].mustRunAfter 'build'
-
-        //Oddly, artifactory plugin doesn't always wire these itself
-        plugins.withId('ivy-publish') {
-          tasks['artifactoryPublish'].dependsOn tasks.withType(GenerateIvyDescriptor)
-        }
-        plugins.withId('maven-publish') {
-          tasks['artifactoryPublish'].dependsOn tasks.withType(GenerateMavenPom)
-        }
-
-        //Auto map publications to artifactory publishing
-        publishing {
-          publications.all { Publication pub ->
-            project.tasks.artifactoryPublish { pubTask ->
-              pubTask.publications(pub.name)
+      /*TODO: Add maven publications:
```

Comment:
I tried to get this to work, but I felt like I started going down a rabbit hole, so I pulled back on it for now. I would like to pair with @stormbeta at some point to see if we can't get this to work.